sample/src/main/kotlin/com/adevinta/spark/sample/backdrop/Backdrop.kt

```diff
@@ -0,0 +1,542 @@
+ * SOFTWARE.
+ */
+
+package com.adevinta.spark.catalog.backdrop
+
+import androidx.compose.animation.core.AnimationSpec
+import androidx.compose.animation.core.TweenSpec
+import androidx.compose.animation.core.animateFloatAsState
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.gestures.Orientation
+import androidx.compose.foundation.gestures.detectTapGestures
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.offset
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.CornerSize
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.saveable.Saver
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.UiComposable
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.graphics.isSpecified
+import androidx.compose.ui.input.nestedscroll.nestedScroll
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.layout.SubcomposeLayout
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.semantics.collapse
+import androidx.compose.ui.semantics.expand
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.offset
+import androidx.compose.ui.zIndex
+import com.adevinta.spark.SparkTheme
+import com.adevinta.spark.catalog.backdrop.BackdropValue.Concealed
+import com.adevinta.spark.catalog.backdrop.BackdropValue.Revealed
+import com.adevinta.spark.components.bottomsheet.PreUpPostDownNestedScrollConnection
+import com.adevinta.spark.components.bottomsheet.SwipeableDefaults
+import com.adevinta.spark.components.bottomsheet.SwipeableState
+import com.adevinta.spark.components.bottomsheet.swipeable
+import com.adevinta.spark.components.snackbars.SnackbarHost
+import com.adevinta.spark.components.snackbars.SnackbarHostState
+import com.adevinta.spark.components.surface.Surface
+import com.adevinta.spark.tokens.contentColorFor
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.launch
+import kotlin.math.max
+import kotlin.math.min
+import kotlin.math.roundToInt
+
+// TODO-scott.rayapoulle.ext (23-15-2023): Remove once M3 has this implementation available
```

Comment:
nit: It could be helpful to keep track of the origin of this file. Do we have a link to the source?